File: src/main/java/com/arnas/klatrebackend/features/invites/InviteRepository.java
package com.arnas.klatrebackend.features.invites;

import java.util.List;

public interface InviteRepository {
    GroupInvite getGroupInviteById(long inviteId);
    long inviteUserToGroup(long userId, long groupId, long senderId);
    List<GroupInvite> getUserInvitesByStatus(long userId, String status);
    int acceptInvite(long inviteId);
    int declineInvite(long inviteId);
    int revokeInvite(long inviteId);


}

File: src/main/kotlin/com/arnas/klatrebackend/features/invites/InviteRepository.kt
package com.arnas.klatrebackend.features.invites

interface InviteRepository {
    fun getGroupInviteById(inviteId: Long): GroupInvite
    fun inviteUserToGroup(userId: Long, groupId: Long, senderId: Long): Long
    fun getUserInvitesByStatus(userId: Long, status: String): List<GroupInvite>
    fun acceptInvite(inviteId: Long): Int
    fun declineInvite(inviteId: Long): Int
    fun revokeInvite(inviteId: Long): Int
}

